blob/fileblob/fileblob.go

```diff
@@ -512,7 +523,102 @@ func (b *bucket) Delete(ctx context.Context, key string) error {
 	return nil
 }
 
+// SignedURL implements driver.SignedURL
 func (b *bucket) SignedURL(ctx context.Context, key string, opts *driver.SignedURLOptions) (string, error) {
-	// TODO(Issue #546): Implemented SignedURL for fileblob.
-	return "", errNotImplemented
+	if b.opts.URLSigner == nil {
+		return "", errors.New("sign fileblob url: bucket does not have an Options.URLSigner")
+	}
+	surl, err := b.opts.URLSigner.URLFromKey(ctx, key, opts)
+	if err != nil {
+		return "", err
+	}
+	return surl.String(), nil
+}
+
+// URLSigner defines an interface for creating and verifying a signed URL for
+// objects in a fileblob bucket. Signed URLs are typically used for granting
+// access to an otherwise-protected resource without requiring further
+// authentication, and callers should take care to restrict the creation of
+// signed URLs as is appropriate for their application.
+type URLSigner interface {
+	// URLFromKey defines how the bucket's object key will be turned
+	// into a signed URL. URLFromKey must be safe to call from multiple goroutines.
+	URLFromKey(ctx context.Context, key string, opts *driver.SignedURLOptions) (*url.URL, error)
+
+	// KeyFromURL must be able to validate a URL returned from URLFromKey.
+	// KeyFromURL must only return the object if if the URL is
+	// both unexpired and authentic. KeyFromURL must be safe to call from
+	// multiple goroutines. Implementations of KeyFromURL should not modify
+	// the URL argument.
+	KeyFromURL(ctx context.Context, surl *url.URL) (string, bool)
+}
+
+// URLSignerHMAC signs URLs by adding the object key, expiration time, and a
+// hash-based message authentication code (HMAC) into the query parameters.
+// Values of URLSignerHMAC with the same secret key will accept URLs produced by
+// others as valid.
+type URLSignerHMAC struct {
+	baseURL   *url.URL
+	secretKey []byte
+}
+
+// NewURLSignerHMAC creates a URLSignerHMAC. A secretKey is required.
+func NewURLSignerHMAC(baseURL *url.URL, secretKey []byte) *URLSignerHMAC {
+	if secretKey == nil {
```

Comment:
Checking the length of the slice is more robust:

```suggestion
	if len(secretKey) == 0 {
```